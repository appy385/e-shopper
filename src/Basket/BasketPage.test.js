import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import BaksetPage from './BasketPage';

describe(BaksetPage.name, () => {
  let mockBasket;
  beforeEach(() => {
    // document.local.href = 'http://localhost/cart';
    mockBasket = {
      'Fruits & Vegatables': [{
        id: 1,
        src: 'https://images.pexels.com/photos/61127/pexels-photo-61127.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500',
        seller: 'Fresho',
        name: 'Banana-Robusta',
        price: 40,
        count: 2,
        category: 'Fruits & Vegatables',
      }],
      'Household Items': [{
        id: 2,
        src: 'https://images.pexels.com/photos/61127/pexels-photo-61127.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500',
        seller: 'Qwerty',
        name: 'table cloth',
        price: 40,
        count: 2,
        category: 'Household Items',
      }],

    };
  });
  afterEach(() => {
    document.close();
  });

  test('should match snapshot', () => {
    const { container } = render(<BrowserRouter><BaksetPage basket={mockBasket} /></BrowserRouter>);
    expect(container).toMatchSnapshot();
  });

  test('should go back to home page', () => {
    render(<BrowserRouter><BaksetPage basket={mockBasket} /></BrowserRouter>);
    const homeLink = screen.getByText('continue shopping');
    fireEvent.click(homeLink);
    expect(document.location.href).toBe('http://localhost/');
  });

  test('should go back to checkout page when basket length>0 ', () => {
    render(<BrowserRouter><BaksetPage basket={mockBasket} /></BrowserRouter>);
    const checkoutLink = screen.getByText('checkout');
    fireEvent.click(checkoutLink);
    expect(document.location.href).toBe('http://localhost/checkout');
  });

  // test('should not go back to checkout page when basket length === 0 ', () => {
  //   render(<BrowserRouter><BaksetPage basket={{}} /></BrowserRouter>);
  //   const checkoutLink = screen.getByText('checkout');
  //   fireEvent.click(checkoutLink);
  //   expect(document.location.href).toBe('http://localhost/cart');
  // });
});
