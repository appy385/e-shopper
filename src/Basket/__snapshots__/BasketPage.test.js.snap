// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasketPage should match snapshot 1`] = `
<div>
  <div
    class="cart"
  >
    <div
      class="cart-table-container"
    >
      <table
        class="cart-table"
      >
        <thead>
          <tr
            class="cart-table-header"
          >
            <th>
              Item Description
            </th>
            <th>
              Unit Price
            </th>
            <th>
              Quantity
            </th>
            <th>
              SubTotal
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="cart-category"
          >
            <td>
              Fruits & Vegatables
            </td>
            <td>
               
            </td>
            <td>
               
            </td>
            <td>
               
            </td>
          </tr>
          <tr
            class="item"
          >
            <td>
              Banana-Robusta
            </td>
            <td>
              40
            </td>
            <td>
              2
            </td>
            <td
              data-testid="item-amount"
            >
              80
            </td>
          </tr>
          <tr
            class="cart-category"
          >
            <td>
              Household Items
            </td>
            <td>
               
            </td>
            <td>
               
            </td>
            <td>
               
            </td>
          </tr>
          <tr
            class="item"
          >
            <td>
              table cloth
            </td>
            <td>
              40
            </td>
            <td>
              2
            </td>
            <td
              data-testid="item-amount"
            >
              80
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="cart-container"
    >
      <a
        href="/"
      >
        <button
          class="cart-back-button"
          type="button"
        >
          continue shopping
        </button>
      </a>
      <div
        class="cart-checkout"
      >
        <p
          class="cart-amount"
        >
          Total: 160
        </p>
        <hr
          class="cart-line"
        />
        <a
          href="/checkout"
        >
          <button
            class="checkout-button"
            type="button"
          >
            checkout
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
